<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QEspLamp</class>
 <widget class="QWidget" name="QEspLamp">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>491</width>
    <height>433</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ESP Lamp controller</string>
  </property>
  <property name="windowIcon">
   <iconset theme="format-fill-color">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Device:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QComboBox" name="modeComboBox">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>SOLID_RARE</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>FADE_ANIM</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="hostLineEdit">
         <property name="text">
          <string>192.168.1.3</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>Submit</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QSpinBox" name="portSpinBox">
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>65535</number>
         </property>
         <property name="value">
          <number>8888</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="color_widgets::ColorWheel" name="wheel" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="color_widgets::ColorPreview" name="preview" native="true"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <property name="sizeConstraint">
        <enum>QLayout::SetMaximumSize</enum>
       </property>
       <item row="2" column="1">
        <widget class="color_widgets::GradientSlider" name="slide_value">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Saturation</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Hue</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="color_widgets::GradientSlider" name="slide_saturation">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="10" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Hex</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Blue</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="color_widgets::GradientSlider" name="slide_red">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="color_widgets::GradientSlider" name="slide_green">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Value</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Green</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Red</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="color_widgets::GradientSlider" name="slide_blue">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QSpinBox" name="spin_hue">
         <property name="wrapping">
          <bool>true</bool>
         </property>
         <property name="maximum">
          <number>359</number>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSpinBox" name="spin_saturation">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSpinBox" name="spin_value">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QSpinBox" name="spin_red">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
       <item row="5" column="2">
        <widget class="QSpinBox" name="spin_green">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
       <item row="6" column="2">
        <widget class="QSpinBox" name="spin_blue">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="3">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="9" column="0" colspan="3">
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="color_widgets::HueSlider" name="slide_hue">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>359</number>
         </property>
        </widget>
       </item>
       <item row="10" column="1" colspan="2">
        <widget class="color_widgets::ColorLineEdit" name="edit_hex">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="showAlpha" stdset="0">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>color_widgets::ColorPreview</class>
   <extends>QWidget</extends>
   <header>color_preview.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>color_widgets::ColorWheel</class>
   <extends>QWidget</extends>
   <header>color_wheel.hpp</header>
   <container>1</container>
   <slots>
    <signal>colorSelected(QColor)</signal>
    <slot>setColor(QColor)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>color_widgets::GradientSlider</class>
   <extends>QSlider</extends>
   <header>gradient_slider.hpp</header>
  </customwidget>
  <customwidget>
   <class>color_widgets::HueSlider</class>
   <extends>color_widgets::GradientSlider</extends>
   <header>hue_slider.hpp</header>
  </customwidget>
  <customwidget>
   <class>color_widgets::ColorLineEdit</class>
   <extends>QLineEdit</extends>
   <header>color_line_edit.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>slide_saturation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QEspLamp</receiver>
   <slot>set_hsv()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>54</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_value</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QEspLamp</receiver>
   <slot>set_hsv()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_red</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QEspLamp</receiver>
   <slot>set_rgb()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>557</x>
     <y>142</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_green</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QEspLamp</receiver>
   <slot>set_rgb()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_blue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QEspLamp</receiver>
   <slot>set_rgb()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>232</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wheel</sender>
   <signal>colorSelected(QColor)</signal>
   <receiver>QEspLamp</receiver>
   <slot>setColorInternal(QColor)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>568</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_saturation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_saturation</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>480</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_saturation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_saturation</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>461</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>416</x>
     <y>71</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_value</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_value</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>480</x>
     <y>91</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_value</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_value</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>480</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>416</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_red</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_red</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_red</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_red</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>162</y>
    </hint>
    <hint type="destinationlabel">
     <x>416</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_green</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_green</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_green</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_green</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>416</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_blue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_blue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>232</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_blue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_blue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel">
     <x>416</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_hue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_hue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>405</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>26</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_hue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_hue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>448</x>
     <y>18</y>
    </hint>
    <hint type="destinationlabel">
     <x>388</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_hue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QEspLamp</receiver>
   <slot>set_hsv()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>17</y>
    </hint>
    <hint type="destinationlabel">
     <x>363</x>
     <y>8</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>QEspLamp</receiver>
   <slot>on_submitClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>435</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>245</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>colorChanged(QColor)</signal>
  <slot>set_rgb()</slot>
  <slot>set_hsv()</slot>
  <slot>setColor(QColor)</slot>
  <slot>setColorInternal(QColor)</slot>
 </slots>
</ui>
